### ISSUE TITLE
value behavior does not type-check nor convert resolved values

### ISSUE THREAD

**Issue by bmomberger-bitovi** [canjs/can-observable-mixin#152]
https://codepen.io/bmomberger-bitovi/pen/OJJGZZq

This should at very least warn in dev mode.  Ideally it should have the same type checking as when the property is set externally.

**Comment by phillipskevin** [comment]
Type checking happens when the property is set, so errors are thrown if you set the `prop` instead of `propValue`.

**Comment by bmomberger-bitovi** [comment]
I don't see the two being mutually exclusive.  We had the same discussion about `type` and `default` (when it was called `value`) ages ago, and I think that the principles still apply here.

**Comment by phillipskevin** [comment]
So you think we should type check twice? Type check when the value is set and type check when `resolve` is called?

**Comment by bmomberger-bitovi** [comment]
Yes, either twice, or only when `resolve` is called when a `value` behavior exists.

Alternatively we could pass `type` as a function parameter to the `value` behavior.  But my gut instinct is that the flexibility to call the type checker/converter whenever isn't more beneficial than the simplicity of leaving it as declarative.
